plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -541,6 +544,65 @@ fu_nordic_hid_cfg_channel_get_bl_name(FuNordicHidCfgChannel *self, GError **erro
 	return TRUE;
 }
 
+static gboolean
+fu_nordic_hid_cfg_channel_get_devinfo(FuNordicHidCfgChannel *self, GError **error)
+{
+	guint8 event_id = 0;
+	g_autoptr(FuNordicCfgChannelMsg) res = g_new0(FuNordicCfgChannelMsg, 1);
+
+	/* query for the devinfo if the board supports it */
+	if (fu_nordic_hid_cfg_channel_get_event_id(self, "dfu", "devinfo", &event_id)) {
+		if (!fu_nordic_hid_cfg_channel_cmd_send(self,
+							"dfu",
+							"devinfo",
+							CONFIG_STATUS_FETCH,
+							NULL,
+							0,
+							error))
+			return FALSE;
+		if (!fu_nordic_hid_cfg_channel_cmd_receive(self, CONFIG_STATUS_SUCCESS, res, error))
+			return FALSE;
+
+		/* check if not set via quirk */
+		if (self->generation != NULL) {
+			g_debug("generation readout '%s' overrides generation from quirk '%s'",
+				g_strndup((const gchar *)res->data, res->data_len),
```

Comment:
same comment about not freeing the g_strndup

Reply:
Reworked a bit